openwisp_firmware_upgrader/base/models.py

```diff
@@ -266,6 +296,32 @@ def create_upgrade_operation(self, batch):
         transaction.on_commit(lambda: upgrade_firmware.delay(operation.pk))
         return operation
 
+    @classmethod
+    def auto_add_device_firmware_to_device(cls, instance, created, **kwargs):
+        # Automatically associate DeviceFirmware to the registered Device
+        if created:
+            if not instance.device.os or not instance.device.model:
+                return
+            if instance.device.model not in REVERSE_FIRMWARE_IMAGE_MAP:
+                return
+
```

Comment:
if many devices exist (eg: hundreds) adding a connection with `auto_add` set to `True` is a bit slow already right now without this receiver, after adding them we risk to make it a bit too slow because the queries we'll be doing here are intensive, which may cause a timeout in the HTTP request.

Can you create a celery task for this?

After `if instance.device.model not in REVERSE_FIRMWARE_IMAGE_MAP:` we can then pass the execution to a background task:

```python
transaction.on_commit(lambda: create_device_firmware.delay(operation.pk))
```

This operation does not need to be synchronous so this would work well.